ksqldb-tools/src/main/java/io/confluent/ksql/tools/migrations/util/CommandParser.java

```diff
@@ -221,16 +220,15 @@ public static ParsedCommand parse(
   /**
    * Validates that the sql statement represented by the list of input tokens
    * (keywords separated whitespace, or strings identified by single quotes)
-   * is not unsupported by the migrations tool. Assumes that tokens have already
-   * been upper-cased.
+   * is not unsupported by the migrations tool.
    *
    * @param sql components that make up the sql statement. Each component is
    *               either a keyword separated by whitespace or a string enclosed
    *               in single quotes. Assumes that tokens have already been upper-cased.
    */
   private static void validateSupportedStatementType(final String sql) {
     final List<String> tokens = Arrays
-            .stream(sql.toUpperCase().split(QUOTED_STRING_OR_WHITESPACE))
```

Comment:
Looks like this code was introduced in https://github.com/confluentinc/ksql/pull/9254 as part of switching from manual parsing of sql statements to using the ksql parser. As part of this switch, this entire method is obsolete and should be replaced with parsed statement type checking post-parser. In other words, there's no benefit trying to validate whether the migration tool supports the statement type before running the sql through the parser -- we should parse first and then perform the type check. The only reason we used to perform the type check first was because we had to, since the manual parsing only supported specific types of statements.

I know your PR is making a local improvement so asking you to take on the removal of this method entirely would be an expansion of scope so I'll leave it up your judgment, but I do think this code would be a good deal cleaner if this method were to be removed entirely.

Reply:
I linked your comment in the javadoc - [6942a23](https://github.com/confluentinc/ksql/pull/9881/commits/6942a2387b1189cfc128f2b776d5c8370657d5de).